components/KiboHamburgerMenu.vue

```diff
@@ -0,0 +1,261 @@
+                v-for="(category, key) in megaMenuCategories"
+                :key="key"
+                :title="category.content.name"
+                :link="localePath(getCatLink(category))"
+                class="side-bar-mega-menu-column"
+              >
+                <SfList>
+                  <SfListItem v-for="child in category.childrenCategories" :key="child.id">
+                    <SfMenuItem
+                      :label="$t(child.content.name)"
+                      class="side-bar-menu-item"
+                      @click="gotoProduct(child)"
+                    />
+                  </SfListItem>
+                </SfList>
+              </SfMegaMenuColumn>
+            </KiboMobileMegaMenu>
+          </div>
+
+          <div class="line-3"></div>
+
+          <!--@slot Use this slot to place content to sticky bottom.-->
+          <div class="sf-sidebar__bottom">
+            <slot name="content-bottom" />
+          </div>
+        </aside>
+      </transition>
+    </div>
+  </div>
+</template>
+<script lang="ts">
+import { onMounted, defineComponent, computed } from "@vue/composition-api"
+import { SfOverlay, SfList, SfMenuItem } from "@storefront-ui/vue"
+import { disableBodyScroll, clearAllBodyScrollLocks } from "body-scroll-lock"
+import { focusTrap, clickOutside } from "@storefront-ui/vue/src/utilities/directives/"
+import { isClient } from "@storefront-ui/vue/src/utilities/helpers"
+import { useCategoryTree, useUiHelpers, categoryGetters } from "@/composables"
+import { useNuxtApp } from "#app"
+
+export default defineComponent({
+  components: {
+    SfList,
+    SfMenuItem,
+    SfOverlay,
+  },
+  directives: { focusTrap, clickOutside },
+  props: {
+    headingLevel: {
+      type: Number,
+      default: 3,
+    },
+    button: {
+      type: Boolean,
+      default: true,
+    },
+    visible: {
+      type: Boolean,
+      default: true,
+    },
+    overlay: {
+      type: Boolean,
+      default: true,
+    },
+    persistent: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  setup(props, context) {
+    const nuxt = useNuxtApp()
+    const app = nuxt.nuxt2Context.app
+    const { categories: allCategories, load: loadCategories } = useCategoryTree()
+    const megaMenuCategories = computed(() => {
+      return categoryGetters.getMegaMenuCategory(allCategories.value)
+    })
+    const { getCatLink } = useUiHelpers()
+
+    const gotoProduct = (child) => {
+      context.emit("closeHamburgerMenu")
+      const linkPath = getCatLink(child)
+      return app.router.push({ path: linkPath })
+    }
+
+    onMounted(async () => {
+      await loadCategories()
+    })
+
+    return {
+      getCatLink,
+      allCategories,
+      megaMenuCategories,
+      gotoProduct,
+    }
+  },
+  data() {
+    return {
+      position: "left",
+      staticClass: null,
+      className: null,
+      active: "",
+      title: "Back",
```

Comment:
Already being passed from the parent. Either use this or use the parent props.

Reply:
fixed.